learning_phase/chat_sample/server.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+)
+
+type server struct {
+	Password          string
+	ClientConnections map[string]*client
+	commands          chan command
+	mu                sync.RWMutex
+}
+
+func newServer(password string) *server {
+	return &server{
+		Password:          password,
+		ClientConnections: make(map[string]*client),
+		commands:          make(chan command),
+	}
+}
+
+func (s *server) run() {
```

Comment:
This function isn't receiving and handling the context cancellation.. and so it won't be able to gracefully terminate (i.e. would be required in case we wanted to perform some operations before exiting this function). This function will be immediately terminated after main() function terminates